woocommerce/woocommerce-ios
# [Mobile Payments] Add badges leading to Set up Tap to Pay

Part of: #9582


## Description


To inform merchants about Tap to Pay being available, we are adding badges to lead merchants to the Set up Tap to Pay on iPhone screen.

These will be shown to merchants whose phone and site could support Tap to Pay, but who have not used it yet.

The badges are added to:
- The `Menu` tab item
- The `Payments` row in the Hub Menu
- The `Set up Tap to Pay on iPhone` row in the Payments Menu 

All badges will remain until the merchant opens the `Set up Tap to Pay on iPhone` screen – after this, they will be removed, regardless of whether the merchant goes through the flow. Alternatively, if they take a Tap to Pay payment without going through the Set up flow, the badges will also be removed.

This PR adds the badges, and some of the removal logic, but is missing three key bits of logic

1. The badges will be shown on any store or device, regardless of Tap to Pay support or previous activity.
2. After removal, the badges will return the next time you open the app.
3. Taking a Tap to Pay payment will not remove the badges

These items will be done in a future PR. This work is feature flagged by `featureFlagService.isFeatureFlagEnabled(.tapToPayBadge)`

## Testing instructions


Open the app
Observe that the badge is shown on the Menu tab
Navigate to `Menu`
Observe that the badge is shown on the Payments row
Navigate to `Payments`
Observe that the badge is shown on the `Set up Tap to Pay on iPhone` row
Tap `Set up Tap to Pay on iPhone`
Tap `Cancel`
Observe that the badges are removed, on the `Menu` tab and the `Set up Tap to Pay on iPhone` row
Tap `<`
Observe that the badge is removed on the `Payments` row.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/2472348/f1daddea-ec6d-44b4-b2fa-d0992cb4db5d

(n.b. I fixed the slight delay in badging the row when the In Person Payments Menu is loaded after recording this video, in 6c81716)

---
